dnf5/context.cpp

```diff
@@ -363,6 +363,12 @@ void Context::Impl::download_and_run(libdnf5::base::Transaction & transaction) {
         base.get_config().get_tsflags_option().set(libdnf5::Option::Priority::RUNTIME, "test");
     }
 
+    if (transaction.test() != libdnf5::base::Transaction::TransactionRunResult::SUCCESS) {
```

Comment:
I think it would be good to perform the test before download, but I don't think it is a good idea to call `transaction.test()` before RPMs are downloaded, because they are required for the test. The code also  does not consider use case with downloadonly option.

Reply:
Could you please help me to figure out where exactly the downloaded RPMs are required for the test to run? 
It isn't really clear for me unfortunately. I don't see anything related to it in `libdnf5::base::Transaction::Impl::_run()`

Also the downloadonly option is apparently unaltered because `Context::Impl::download_and_run()` isn't called for `dnf5 download package_name`. I'm not that fluent in DNF though... feel free to correct me here. 

It probably would make way more sense to check for the particular error `TransactionRunResult::ERROR_LOCK` and show the error message only in case of this error... 

Should we rather have something like `transaction.lock_test()` that only check for permissions and don't rely on the whole `transaction.test()` functionality (which as you said requires RPMs to be downloaded). 

P.S.
Rebased on top of main as `download_and_run` just got some updates.